### PR TITLE
Remove writing buffers

### PR DESCRIPTION
**Pull request by pascutto**
This seems to be a big perf regression with my homemade benchs; but checking with the upstream ones.

### CODE REVIEW

File: src/unix/index_unix.ml [hunk 1]
```diff
@@ -26,64 +26,41 @@ let current_version = "00000001"
 module Stats = Index.Stats
 
 module IO : Index.IO = struct
-  external ( ++ ) : int64 -> int64 -> int64 = "%int64_add"
-
-  external ( -- ) : int64 -> int64 -> int64 = "%int64_sub"
+  let ( ++ ) = Int64.add
```
Comment (samoht) [inline src/unix/index_unix.ml:29]: this is allocating too much (keep the externals)

File: src/unix/index_unix.ml [hunk 2]
```diff
@@ -26,64 +26,41 @@ let current_version = "00000001"
...
-    Raw.close t.raw
-
-  let auto_flush_limit = 1_000_000L
+  let close t = Raw.close t.raw
 
   let append t buf =
+    t.flush_callback ();
     if t.readonly then raise RO_not_allowed;
-    Buffer.add_string t.buf buf;
+    Raw.unsafe_write t.raw ~off:(t.header ++ t.offset) buf;
```
Comment (samoht) [inline src/unix/index_unix.ml:58]: This will cause a lot of short writes. Are you sure it's ok?

Comment (pascutto) [reply src/unix/index_unix.ml:58]: It's not, I want to benchmark how much it is not :)

Comment (samoht) [reply src/unix/index_unix.ml:58]: sounds good :-)

**Comment by pascutto** [comment]
Experiment finished, sorry for the noise :)

**Pull request closed without merging**